trin-core/src/portalnet/overlay_service.rs

```diff
@@ -357,17 +366,24 @@ impl OverlayService {
     }
 
     /// Returns list of nodes closer to content than self, sorted by distance.
-    async fn find_nodes_close_to_content(&self, content_key: Vec<u8>) -> Vec<SszEnr> {
+    async fn find_nodes_close_to_content(
+        &self,
+        content_key: Vec<u8>,
+    ) -> Result<Vec<SszEnr>, OverlayRequestError> {
         let self_node_id = self.local_enr().await.node_id();
-        let self_distance = xor_two_values(&content_key, &self_node_id.raw().to_vec());
+        let self_distance = match xor_two_values(&content_key, &self_node_id.raw().to_vec()) {
+            Ok(val) => val,
+            Err(msg) => return Err(OverlayRequestError::InvalidRequest(msg.to_string())),
```

Comment:
Maybe a bit more context here? Like `format!("Could not find distance from node, because content key is malformed: {}", msg.to_string())`